charts/portworx/templates/storage-cluster.yaml

```diff
@@ -91,11 +97,23 @@ spec:
       {{- range $index, $name := $driveNames }}
       - {{ $name }}
       {{- end }}
+    {{- if and (ne $kvdbDevice "none") (not (hasPrefix "/" $kvdbDevice)) }}
+    kvdbDeviceSpec: {{ $kvdbDevice }}
+    {{- end }}
+    {{- if ne (.Values.maxStorageNodesPerZone | int) 0 }}
+    maxStorageNodesPerZone: {{ .Values.maxStorageNodesPerZone }}
+    {{- end }}
 {{- else }}
-  {{- if .AKSInstall }}
+  {{- if $AKSInstall }}
   cloudStorage:
     deviceSpecs:
       - type=Premium_LRS,size=150
+    {{- if and (ne $kvdbDevice "none") (not (hasPrefix "/" $kvdbDevice)) }}
```

Comment:
There is kvdb device in storage section, it should not be used if it's cloud storage format. technically users should not configure device in storage format and kvdb device in cloud storage format, but better to check the kvdb device format is storage format to be safe.

    {{- if ne $kvdbDevice "none" }}
    kvdbDevice: {{ $kvdbDevice }}
    {{- end }}